resources/giwscripts/giwtypes/eigen3.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module is concerned with the analysis of each variable found by the
+debugger, as well as identifying and describing the buffers that should be
+plotted in the ImageWatch window.
+"""
+
+import re
+
+from giwscripts import symbols
+from giwscripts.giwtypes import interface
+
+import numpy as np
+
+# we need the gdb module as we need to call methods
+# for finding the data, width and height of
+# Eigen::Matrix or Eigen::Map
+import gdb
+
+
+class EigenXX(interface.TypeInspectorInterface):
+    """
+       Implementation for inspecting Eigen::Matrix and Eigen::Map
+       """
+    def get_buffer_metadata(self, obj_name, picked_obj, debugger_bridge):
+        """
+        Gets the buffer meta data from types of the Eigen Library
+        Note that it only implements single channel matrix display,
+        which should be quite common in Eigen.
+        """
+
+        # first we need to parse the underlying type and size
+        # from our picked_ojb type name
+        type_and_size_expression = picked_obj.type.name
+        m = re.search('(float|short|double|int), (\d+), (\d+), (\d+)',type_and_size_expression)
+        if m:
+            current_type = m.group(1)
+            height = int(m.group(2))
+            width = int(m.group(3))
+            is_row_major = int(m.group(4))
+        else:
+            raise Exception('Invalid or not implemented type!')
+
+        if not is_row_major:
+            raise Exception('You are trying to display a col major matrix. This is not supported')
+
+        # assing the GIW type according to underlying type
+        if current_type == 'short':
+            type_value = symbols.GIW_TYPES_INT16
+        elif current_type == 'float':
+            type_value = symbols.GIW_TYPES_FLOAT32
+        elif current_type == 'double':
+            type_value = symbols.GIW_TYPES_FLOAT64
+        elif current_type == 'int':
+            type_value = symbols.GIW_TYPES_INT32
+
+        # differentiate between Map and Matrix in handling
+        if 'Map' in str(picked_obj.type):
+            str_data = obj_name + '.m_data'
+            buffer = debugger_bridge.get_casted_pointer(current_type, gdb.parse_and_eval(str_data))
```

Comment:
Isn't it possible to access the `m_data` field by callig `picked_obj['m_data']` instead? If so, that would be preferable since it would make the Eigen type script agnostic to debugger frontend, so the gdb module would not be necessary. The same should also apply to `picked_obj['m_storage']` below.

I'm trying to avoid using the gdb module explicitly outside the gdbbridge module since that would make it possible to extend the plugin to other debuggers in the future (like lldb) with little effort.